strawberryfields/backends/states.py

```diff
@@ -814,6 +815,43 @@ def expand_dims(op, n, modes):
 
         return mean, var
 
+    def number_expectation(self, modes):
+        """
+        Calculates the expectation value of the product of the number operators of the modes.
+        Args:
```

Comment:
Perhaps good to add more details here:
```suggestion
        This method computes the analytic expectation value
        :math:`\langle \hat{n}_{i_0} \hat{n}_{i_1}\dots \hat{n}_{i_{N-1}}\rangle`
        for a (sub)set of modes `[i_0, i_1, \dots, i_{N-1}]` of the system.

        Args:
```